components/dialog/dialog-mixin.js

```diff
@@ -561,6 +561,8 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			await this.updateComplete;
 			this._height = this._getHeight();
 			await this.updateComplete;
+		} else {
+			this._width = 0;
```

Comment:
It looks like `d2l-dialog-fullscreen` would fall into this block. Presumably if that's a problem, visual-diffs would be complaining about it. It appears to have [this line](https://github.com/BrightspaceUI/core/blob/main/components/dialog/dialog-fullscreen.js#L217) that overrides this `_width` value.

Reply:
Is there any worry about this affecting [this height calculation](https://github.com/BrightspaceUI/core/blob/main/components/dialog/dialog-mixin.js#L262)?

Reply:
Yes, full screen dialogs do fall into that but you're correct in that they just override the width anyway. I'm not worried about the height calculation because `_getHeight()` (where that `fullHeight` thing is checked) is only called when `autoSize` is in play.

The main reason this feels very safe to me is that `this._width` is initialized to zero in the constructor already, which is why the bug doesn't exist if you initially load the page up in a mobile viewport size. It's only once it's loads in a desktop size, width gets set, then you go to mobile that things started to fail. This simply resets it back to its original value when that happens.